standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -258,6 +258,7 @@ public CompactionInfo findNextToCompact(FindNextCompactRequest rqst) throws Meta
           sb.append("\"CQ_POOL_NAME\" IS NULL OR  \"CQ_ENQUEUE_TIME\" < (")
             .append(getEpochFn(dbProduct)).append(" - ").append(poolTimeout).append(")");
         }
+        sb.append(" ORDER BY CQ_ID ASC");
```

Comment:
`sb.append(" ORDER BY \"CQ_ID ASC\"");`

or you can have issues with Postgresql.

Reply:
Good point! cc: @veghlaci05

Reply:
@aturoczy I will re-do this change, already splitted into two (the flaky test fix no separated). but currently the revert change has failures.

The majority of the tests are running on Derby unfortunately.